src/voice-search/voice-search.ts

```diff
@@ -0,0 +1,162 @@
+import { BaseWidget } from '../base-widget';
+import { NgAisInstantSearch } from '../instantsearch/instantsearch';
+import { noop } from '../utils';
+
+type TemplateContext = {
+  status: string;
+  transcript: string;
+  isSpeechFinal: boolean;
+  errorCode?: string;
+  isBrowserSupported: boolean;
+  isListening: boolean;
+};
+
+type State = {
+  isBrowserSupported: boolean;
+  isListening: boolean;
+  toggleListening: () => void;
+  refine: (query: string) => void;
+  voiceListeningState: VoiceListeningState;
+  templateContext: TemplateContext;
+};
+
+@Component({
+  selector: 'ais-voice-search',
+  template: `
+    <div [class]="cx()">
+      <button
+        type="button"
+        [class]="cx('button')"
+        [title]="state.isBrowserSupported ? buttonTitle : disabledButtonTitle"
+        [disabled]="!state.isBrowserSupported"
+        (click)="handleClick($event)"
+      >
+        <ng-container *ngTemplateOutlet="button ? button : defaultButton; context: state.templateContext"></ng-container>
+      </button>
+      <div [class]="cx('status')">
+        <ng-container *ngTemplateOutlet="status ? status : defaultStatus; context: state.templateContext"></ng-container>
+      </div>
+    </div>
+
+    <ng-template #defaultButton let-status="status" let-errorCode="errorCode" let-isListening="isListening">
+      <svg
+        xmlns='http://www.w3.org/2000/svg'
+        width='16'
+        height='16'
+        viewBox='0 0 24 24'
+        fill='none'
+        stroke='currentColor'
+        strokeWidth='2'
+        strokeLinecap='round'
+        strokeLinejoin='round'
+      >
+        <ng-container *ngIf="isNotAllowedError(); then errorSvgContent else normalSvgContent">
+        </ng-container>
+        <ng-template #errorSvgContent>
+          <line x1="1" y1="1" x2="23" y2="23"></line>
+          <path d="M9 9v3a3 3 0 0 0 5.12 2.12M15 9.34V4a3 3 0 0 0-5.94-.6"></path>
+          <path d="M17 16.95A7 7 0 0 1 5 12v-2m14 0v2a7 7 0 0 1-.11 1.23"></path>
+          <line x1="12" y1="19" x2="12" y2="23"></line>
+          <line x1="8" y1="23" x2="16" y2="23"></line>
+        </ng-template>
+        <ng-template #normalSvgContent>
+          <path
+            d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
+            [attr.fill]="isListening ? 'currentColor' : 'none'"
+          ></path>
+          <path d="M19 10v2a7 7 0 0 1-14 0v-2"></path>
+          <line x1="12" y1="19" x2="12" y2="23"></line>
+          <line x1="8" y1="23" x2="16" y2="23"></line>
+        </ng-template>
+      </svg>
+    </ng-template>
+    <ng-template #defaultStatus let-transcript="transcript">
+      <p>{{transcript}}</p>
+    </ng-template>
+  `,
+})
+export class NgAisVoiceSearch extends BaseWidget implements OnInit {
+  @ContentChild('button') button: TemplateRef<ElementRef>;
+  @ContentChild('status') status: TemplateRef<ElementRef>;
+
+  @Input() public searchAsYouSpeak: boolean = false;
```

Comment:
Since this is directly passed to the connector, I would not put a default value here and rather let the connector use its own default value.
The rational behind it it that it's best to maintain theses default values in one single place to avoid differences.

Reply:
Unfortunately there is the default value [on the widget level in the vanilla](https://github.com/algolia/instantsearch.js/blob/develop/src/widgets/voice-search/voice-search.tsx#L82). The widget in vanilla and the component in angular are the outermost layer that intefacing users. IMO the default value better be there. What do you think?

Reply:
Yes. I think we agree, but let me reformulate just in case.
What i'm suggesting is to have no default value in Angular InstantSearch:
```ts
 @Input() public searchAsYouSpeak: boolean;
```
I think that's what you meant too, right?

Reply:
If there's no default value for that, it means it should be optional like this

```js
 @Input() public searchAsYouSpeak?: boolean;
```

and when `undefined` is passed to the connector, the connector will use its own default value. That's what you're saying, right?

Reply:
yes, sorry I forgot the `?`. That's what I mean.

Reply:
That's a fair point. In InstantSearch.js, we should have defaulted the `searchAsYouSpeak` value [in the connector](https://github.com/algolia/instantsearch.js/blob/ab6d8ba1c438c98813161f7505fded90c01167d8/src/connectors/voice-search/connectVoiceSearch.ts#L74) so that users creating their own widget based on `connectVoiceSearch` don't have to explicitly set the value (this is what we do in other connectors).

Reply:
Okay, I will make a PR for this in instantsearch.js first and when it's released I will bring it back to angular-IS.

Reply:
Now it's optional: 2588735
When https://github.com/algolia/instantsearch.js/commit/9d3e91bc2e848f887c4b49f081a355bdc5c9ef0b is published, angular-IS will update the version of IS.js and this thread can be resolved then.

Reply:
6f26093
IS.js@3.5.3 updated